app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+                      List.empty
+                    )
+                  } else {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      supportingEvidences
+                    )
+                  }
+                case CompleteUploadSupportingEvidenceAnswers(
+                    oldDoYouWantToUploadSupportingEvidenceAnswer,
+                    supportingEvidences
+                    ) =>
+                  if (newDoYouWantToUploadSupportingEvidenceAnswer =!= oldDoYouWantToUploadSupportingEvidenceAnswer) {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      List.empty
+                    )
+                  } else {
+                    IncompleteUploadSupportingEvidenceAnswers(
+                      Some(newDoYouWantToUploadSupportingEvidenceAnswer),
+                      supportingEvidences
+                    )
+                  }
+              }
+
+              val newDraftReturn = fillingOutReturn.draftReturn match {
+                case s @ DraftSingleDisposalReturn(
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _
+                    ) =>
+                  s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+                case m @ DraftMultipleDisposalsReturn(
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _,
+                      _
+                    ) =>
+                  m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+              }
+
+              val result = for {
+                _ <- returnsService
+                      .storeDraftReturn(
+                        newDraftReturn,
+                        fillingOutReturn.subscribedDetails.cgtReference,
+                        fillingOutReturn.agentReferenceNumber
+                      )
+                _ <- EitherT(
+                      updateSession(sessionStore, request)(
+                        _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                      )
+                    )
+              } yield ()
+
+              result.fold({ e =>
+                logger.warn("Could not update session", e)
+                errorHandler.errorResult()
+              }, _ => Redirect(routes.UploadSupportingEvidenceController.checkYourAnswers()))
+
+            } else {
+              val result = for {
+                _ <- upscanService.removeAllFiles(DraftReturnId(draftReturnId.toString))
```

Comment:
you should be able to tell if there are actually any files to delete based on the current answers in session. You can skip a call if you know there's nothing to delete

Reply:
will be addressed after 6th April